src/arsenic/services.py

```diff
@@ -36,16 +40,18 @@ async def subprocess_based_service(
         session = ClientSession()
         closers.append(session.close)
         count = 0
-        while True:
-            try:
+
+        async def wait_service():
+            # Wait for service with exponential back-off
+            for i in range(-10, 9999):
```

Comment:
```suggestion
            for i in range(-10, 0):
```

Reply:
If we only go up to 0 then that limits the timeout to ~1 second as you demonstrated already. After this then the current code would just return regardless of whether the service was up or not.

I can foresee some stalls lasting longer than this - particularly in a busy environment where there are lots of other things going on. In my specific use case I have many instances of this going on for parallel test and the webdriver start seems to be one of the most intensive parts - if a few too many of these happen at at same time then, blam, timeout.

I don't see the issue in leaving the max. exponent as some large number. If someone wants to try and wait until the heat death of the universe then that's up to them - at least we won't ever leave the wait_service() coro too early by accident.